smortex/puppetlabs-postgresql
# arch service status

- Use OS major version numbers in globals
- Remove redundant selector for Archlinux
- Update to pdk-templates and puppet-module-gems 1.0.0
- Add the FrozenStrings magic comment
- Apply remaining rubocop fixes
- Remove Style/FrozenStringLiteralComment todo; update to latest templates again
- Fix typos and frozen string modification
- Fix Naming/AccessorMethodName issue
- Update to newest template version
- More FrozenString fixes
- Remove translations
- Revert the faulty gem exclusion for now
- Set default PostgresQL version for FreeBSD
- Release version 6.9.0
- (maint) Update github actions
- Fix postgresql::sql task when password is not set
- (IAC-1136) disable centos8 on travis
- (bug) reload the service after the deletion of old unit file
- Also perform systemd daemon-reload on Puppet 6.1+
- Move the Template uses comment to the actual template
- Improve systemctl daemon-reload comment wording
- Release version 6.10.0
- Update pdk-templates for gitpod and codespaces support
- Release version 6.10.1
- Remove Puppet 5 testing
- Bump minimal puppet version to 6.0.0 in metadata.json
- Update .sync.yml
- pdk update
- Add DNF module management
- Add auto_release.yml to sync.yml
- Clean up postgresql validator for better perf and robustness
- Fix quotes to avoid shell injection
- fix shell command escaping in sql task and spec_helper
- update unit tests
- Implement a `sensitive` param for postgresql_psql
- Add more sensitive protection and testing
- Release version 6.10.2
- (maint) Update major dependencies prior to release
- (MAINT) Remove RHEL 5 OS family support
- (MAINT) Remove SLES 11 support
- (MAINT) pdk update
- Release version 7.0.0
- Ensure port is a string in psql command
- Release version 7.0.1
- (MODULES-10957)override the set_sensitive_parameters method
- Release version 7.0.2
- Replace Travis and Appveyor with Github Actions
- PDK Update for release prep workflow
- Add new common Yum repo which contains add-ons
- Release prep v7.0.3
- Release prep v7.1.0
- Support grants and revokes for PUBLIC pseudo role
- (maint) Running a pdk update to consume latest pdk-template changes
- added default version for fedora 34
- Release prep v7.2.0
- Autorequire the service in postgresql_psql
- (maint) pdksync - PDK Update
- Implement default privileges changes
- (maint) update test with the changes from Autorequire the service
- (maint) Updating changelog generator config
- Do not add version component to repo definition
- (maint) Updating CONTRIBUTING guidelines
- (maint) - Set max_issues for changelog generator to 500
- Release prep v7.3.0
- Reflect change of default in ensure_packages
- Remove use of spec_helper_acceptance in unit tests
- (IAC-1709) - Add Support for Debian 11
- (IAC-1727) - Set default Postgresql version for Debian 11
- Use Puppet-Datatype Sensitive for Passwords
- drop code for Debian 6/7 and Ubuntu 10.04/12.04
- Release prep v7.4.0
- (maint) Allow stdlib 8.0.0
- Release prep v7.4.1
- Fix changing default encoding
- (IAC-1598) - Remove Support for Debian 8
- Inline file contents in the catalog
- Release prep v7.5.0
- Drop further code for Debian 6 and Ubuntu 10
- add service_status for Ubuntu 18.04 and later
- Support target_role in default_privileges
- Support setting default_privileges on all schemas
- (IAC-1751) - Add Support for Rocky 8
- pdk update
- (IAC-1776) - Split Acceptance test workflows in two
- (IAC-1752) - Add Support for AlmaLinux 8
- (IAC-1775) Update workflows
- (IAC-1775) Updates test setup
- not installing postgresql-contrib if not need to
- fixing unit testing contrib package
- fix: Layout/EmptyLinesAroundBlockBody: Extra empty line detected at block body end
- Support schema privileges in default_privileges
- add tests for new implementation contrib
- add support for Ubuntu Hirsute and Impish
- add scram-sha-256 support
- fix table grant with schema
- add new specific variable for onlif commands
- Install python3-psycopg2 in RHEL/CentOS 8
- Allow systemd to mask postgresql service file
- add default version for Fedora 35
- change md5 per default
- fix order
- fix optional
- fix optional
- Fix puppet-lint violations
- Explicitly pass in a boolean for sensitive
- Run rubocop autocorrect
- Manual Rubocop fix
- Follow rspec-puppet path conventions
- Write facts as shared examples and move to Debian 11
- fix spec postgresql_password with scram-sha-256
- Fix spec tests
- (IAC-1787) Remove Support for CentOS 6
- Release prep v8.0.0
- Remove unused variable in reload.pp
- pdksync_heads/main-0-gf3911d3
- (GH-iac-334) Remove Support for Ubuntu 16.04
- (GH-iac-334) Remove Support for Ubuntu 14.04
- README.md: correct postgresql_conn_validator example
- (MAINT) Add labeller and stale GHA workflows
- (MAINT) Fixes no new line at EOF
- (GH-C&T-9) Temporarily disable syntax checks
- Use systemctl reload on EL 7 and higher
- "This commit changes the workflow trigger for pull requests to pull_request_target"
- (maint) PDK Update
- (GH-cat-9) syntax:hiera:yaml fixes
- (GH-cat-9) syntax:hiera:yaml exclusions added
- (GH-cat-9) Rubocop fixes
- (MAINT) Stale-bot config/msg update
- Config update
- Provide service_provider for all supported OSes
- Use service provider to determine service commands
- Add Red Hat Enterprise Linux 9 support
- added fedora 36
- Add version for Ubuntu 22.04
- Respect $service_status on Red Hat-based distros
- Fix service status detection
- (FM-8971) allow deferred function for role pwd
- update postgresql package version
- bump SLES 15 version
- Release prep v8.1.0
- Ensure multiple postgresql::server::recovery resources can be defined
- Fixed tests
- (MODULES-11251) Add support for backup provider "pg_dump"
- (GH-cat-11) Certify Support for Ubuntu 22.04
- Only require password when used
- Add relevant exclusion to pr_test.yml
- Add relevant exclusion to nightly.yml
- Updating to use apt version 9
- added rspec on specific schema name only if object exists
- fix lint
- Release prep v8.2.0
- (GH-1360) Reverting REFERENCE.md changes
- Fix puppet-strings documentation
- Release prep v8.2.1
- (CONT-130) Dropping Debian 9 Support
- (PDKSync) Removal of puppet_module_gems
- (PDKSync) .sync.yml update
- (CONT-189) Remove support for RedHat6
- (CONT-189) Remove support for OracleLinux6
- (CONT-189) Remove support for Scientific6
- (CONT-189) Remove RHEL 6 from Spec tests
- (maint) Codebase Hardening
- Rewrite dependency installation using native Puppet
- pg_hba_rule does not properly verify address parameter
- pg_hba_rule: Move `type` datatype to own type
- pg_hba_rule: Validate userinput in postgresql::server
- (MAINT) Remove stalebot workflow
- (CONT-494) Pin github_changelog_generator gem version
- (CONT-494) PDK Update
- Add multi instance support, refactoring config.pp
- remove debian 8 and 9 corpses,
- Ubuntu 14/16/17: Drop code leftovers
- Add multi instance support, refactoring initdb.pp
- Add multi instance support, refactoring late_initdb.pp
- apt::source: configure repo only for current architecture
- bugfix - client and server package names were swapped around
- .fixtures.yml: Cleanup Puppet 5 leftovers
- Add multi instance support, refactoring service.pp (4/x)
- Add multi instance support, refactoring password.pp (5/x)
- Add multi instance support, refactoring reload.pp (6/x)
- postgresql instances move and rename defines (7/x)
- postgresql instances fix parameter indention server::instance::passwd (8/x)
- Adding mend file
- remove manual systemd drop-in file handling
- remove manual systemd drop-in file handling
- convert ERB templates to EPP
- deprecate old code managing sysconfig files
- (CONT-361) Syntax update
- Parameter documentation rule
- (MAINT) - update pdk release image
- "This change pins the puppetlabs-puppet_agent module to v4.12.1. Previosuly the fixutre was configured to pull from main. Given the recent changes when moving towards puppet8 main is unsafe."
- Fix default value for $service_status on ArchLinux
